sdk/metric/provider.go

```diff
@@ -0,0 +1,96 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric // import "go.opentelemetry.io/otel/sdk/metric"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/metric"
+)
+
+// MeterProvider handles the creation and coordination of Meters. All Meters
+// created by a MeterProvider will be associated with the same Resource, have
+// the same Views applied to them, and have their produced metric telemetry
+// passed to the configured Readers.
+type MeterProvider struct {
+	// TODO (#2820): implement.
+}
+
+// Compile-time check MeterProvider implements metric.MeterProvider.
+var _ metric.MeterProvider = (*MeterProvider)(nil)
+
+// NewMeterProvider returns a new and configured MeterProvider.
+//
+// By default, the returned MeterProvider is configured with the default
+// Resource and no Readers.
+func NewMeterProvider(options ...Option) *MeterProvider {
+	// TODO (#2820): implement.
+	return &MeterProvider{}
+}
+
+// Meter returns a Meter with the given name and configured with options.
+//
+// The name should be the name of the instrumentation library creating
+// telemetry. This name may be the same as the instrumented code only if that
+// code provides built-in instrumentation.
+//
+// If name is empty, the default (go.opentelemetry.io/otel/sdk/meter) will be
+// used.
+//
+// Calls to the Meter method after Shutdown has been called will return Meters
+// that perform no operations.
+//
+// This method is safe to call concurrently.
+func (mp *MeterProvider) Meter(name string, options ...metric.MeterOption) metric.Meter {
+	// TODO (#2821): ensure this is concurrent safe.
+	// TODO: test this is concurrent safe.
+	// TODO (#2821): register and track the created Meter.
+	return &meter{}
+}
+
+// Flush flushes all pending telemetry.
+//
+// This method honors the deadline or cancellation of ctx. An appropriate
+// error will be returned in these situations. There is no guaranteed that all
+// telemetry be flushed or all resources have been released in these
+// situations.
+//
+// This method is safe to call concurrently.
+func (mp *MeterProvider) Flush(ctx context.Context) error {
+	// TODO (#2820): implement.
+	// TODO: test this is concurrent safe.
+	return nil
+}
+
+// Shutdown shuts down the MeterProvider flushing all pending telemetry and
+// releasing any held computational resources.
+//
+// This call is idempotent. The first call will perform all flush and
+// releasing operations. Subsequent calls will perform no action.
+//
+// Calls to the Meter method after Shutdown has been called will return Meters
+// that perform no operations.
```

Comment:
Will it also shutdown meters, instruments, views, and anything else that it might create?

Reply:
Meters, instruments, and views don't have shutdown operations defined, though, in the spec. I think we can say this will release resources held by the SDK itself and Shutdown each reader/exporter. (I refuse to be specific about what is a reader and what is an exporter!)

Reply:
Right. I guess this could be restated that all Meter's that will be and _have been_ returned by the MeterProvider become no-ops. The existing ones will no longer have their telemetry transmitted.

@MadVikingGod does this address the issue?

```suggestion
// Measurements made by instruments from meters this MeterProvider created will not be exported after Shutdown is called.
```

Reply:
LGTM.